framework/src/problems/ExternalProblem.C

```diff
@@ -0,0 +1,31 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+// MOOSE includes
+#include "ExternalProblem.h"
+#include "NonlinearSystem.h"
+#include "AuxiliarySystem.h"
+
+template <>
+InputParameters
+validParams<ExternalProblem>()
+{
+  InputParameters params = validParams<FEProblemBase>();
+  params.set<bool>("skip_nl_system_check") = true;
+
+  return params;
+}
+
+ExternalProblem::ExternalProblem(const InputParameters & parameters) : FEProblemBase(parameters)
+{
+  _nl = std::make_shared<NonlinearSystem>(*this, "nl0");
```

Comment:
I thought we should have a `ExternalNonlinearSystem`?

Reply:
No, the correct solution would be to not have a NonlinearSystem at all for external applications. They should never need one (e.g. definition of external). However, MOOSE is pretty far away from supporting a simulation without a nonlinear system right now. The system has to be there but it can be empty.

Reply:
The good news is that I added another Boolean so that it's possible to have an empty nonlinear system. Previously, we required a single dummy variable but that's no longer an issue as long as you don't attempt to solve (which is also part of this PR).

Reply:
I'm sorry but this is an absolutely terrible idea.... Like the worst. There are zero differences between NonlinearSystem and ExternalNonlinearSystem. So why create another class?